src/System.Collections/src/System/Collections/Generic/SortedSet.TreeSubSet.cs

```diff
@@ -310,11 +311,27 @@ private void VersionCheckImpl()
                 {
                     root = _underlying.FindRange(_min, _max, _lBoundActive, _uBoundActive);
                     version = _underlying.version;
+                }
+            }
+
+            internal override void VersionCheckCount()
+            {
+                VersionCheckImpl();
+
+                if (versionCount != _underlying.version)
```

Comment:
That sounds like a good idea. The only issue is that we have no way of knowing if the current Count property of TreeSubSet is up to date. As a result very time the Count property is accessed (either through within the code or by the end user), Count will always be forcefully updated.

How about adding the updateCount bool like you mentioned (still gets rid of the VersionCheckCount method) but also keeping versionCount variable and renaming it to something easier to understand?

Reply:
How about adding the updateCount bool like you mentioned (still gets rid of the VersionCheckCount method) but also keeping versionCount variable and renaming it to something easier to understand?

That is true, I'm fine with keeping it the way it is, but please rename the variable to something easier to understand and add a comment in the new method just so that people that reads that understand why we need an extra variable and what method is doing.